spec/mem/pointer.md

```diff
@@ -172,18 +189,19 @@ impl PtrType {
     }
 
     pub fn addr_valid(self, addr: Address) -> bool {
-        if let Some(layout) = self.safe_pointee() {
+        if let Some(pointee) = self.safe_pointee() {
             // Safe addresses need to be non-null, aligned, and not point to an uninhabited type.
             // (Think: uninhabited types have impossible alignment.)
-            addr != 0 && layout.align.is_aligned(addr) && layout.inhabited
+            // FIXME(UnsizedTypes): Compute the alignment, don't assume it is sized.
+            addr != 0 && pointee.layout.expect_align("FIXME").is_aligned(addr) && pointee.inhabited
         } else {
             true
         }
     }
```

Comment:
Do you agree that this function should check the dynamic alignment and therefore needs the full pointer?

Reply:
Ah, good question... yes we probably want that. But I am not sure if it is better to pass in the metadata as an extra argument, or pass in the full pointer. You can check out how this function is called, that should help with this.